api/src/main/java/org/openmrs/module/referencedemodata/patient/DemoPatientGenerator.java

```diff
@@ -332,40 +387,17 @@ private Encounter createVisitNote(Patient patient, Date encounterTime, Location
 		
 		// TODO 5% of diagnoses should be non-coded.
 		if (allDiagnoses.size() > 0) {
-			createDiagnosisObsGroup(true, patient, visitNote, encounterTime, location);
+			getDemoDiagnosisGenerator().createDiagnosis(true, patient, visitNote, location, getAllDiagnoses());
 			
 			if (flipACoin()) {
```

Comment:
Maybe instead of this we generate between 1 and 3 diagnoses per patient? Should diagnoses take into account existing conditions?

Reply:
>should diagnoses take into account existing conditions?

Yes because diagnosis has a condition field being referenced for platform 2.2+